src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -2494,6 +2494,747 @@ End Class
+        M5((1, 1))
+        M6({1})
+        M7({(1, 1)})
+    End Sub
+End Class
+    </file>
+</compilation>,
+options:=TestOptions.ReleaseExe.WithOverflowChecks(True), additionalRefs:=s_valueTupleRefs)
+
+            AssertTheseDiagnostics(comp, <expected></expected>)
+
+            CompileAndVerify(comp, expectedOutput:=
+"Byte
+Byte
+Byte
+Byte
+Byte")
+        End Sub
+
+        <Fact>
+        <WorkItem(14473, "https://github.com/dotnet/roslyn/issues/14473")>
+        Public Sub NarrowingFromNumericConstant_07()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Option Strict On
+
+Public Class C
+
+    Shared Sub M1 (x as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub M2 (x as Byte(), y as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub M3 (x as (Byte, Byte))
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub M4 (x as (Byte, Byte), y as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub M5 (x as Byte?)
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub M6 (x as (Byte?, Byte?))
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub M7 (x as (Byte, Byte)?)
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub M8 (x as (Byte, Byte)?, y as Byte)
+        System.Console.WriteLine("Byte")
+    End Sub
+
+    Shared Sub Main()
+        M1(300)
+        M2({ 300 }, 300)
+        M3((300, 300))
+        M4((300, 300), 300)
+
+        M5(70000)
+        M6((70000, 70000))
+        M7((70000, 70000))
+        M8((300, 300), 300)
+    End Sub
+End Class
+    </file>
+</compilation>,
+options:=TestOptions.ReleaseExe.WithOverflowChecks(False), additionalRefs:=s_valueTupleRefs)
+
+            AssertTheseDiagnostics(comp, <expected></expected>)
+
+            CompileAndVerify(comp, expectedOutput:=
+"Byte
+Byte
+Byte
+Byte
+Byte
+Byte
+Byte
+Byte")
+        End Sub
+
+        <Fact>
+        Public Sub NarrowingFromNumericConstant_08()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Option Strict Off
+
+Public Class C
+
+    Shared Sub Main()
+        Dim x00 As (Integer, Integer) = (1, 1)
+        Dim x01 As (Byte, Integer) = (1, 1)
+        Dim x02 As (Integer, Byte) = (1, 1)
+        Dim x03 As (Byte, Long) = (1, 1)
+        Dim x04 As (Long, Byte) = (1, 1)
+        Dim x05 As (Byte, Integer) = (300, 1)
+        Dim x06 As (Integer, Byte) = (1, 300)
+        Dim x07 As (Byte, Long) = (300, 1)
+        Dim x08 As (Long, Byte) = (1, 300)
+        Dim x09 As (Long, Long) = (1, 300)
+        Dim x10 As (Byte, Byte) = (1, 300)
+        Dim x11 As (Byte, Byte) = (300, 1)
+        Dim one As Integer = 1
+        Dim x12 As (Byte, Byte, Byte) = (one, 300, 1)
+        Dim x13 As (Byte, Byte, Byte) = (300, one, 1)
+        Dim x14 As (Byte, Byte, Byte) = (300, 1, one)
```

Comment:
Yes, TupleConvertedType01 covers that. The ```?``` inside the tuple type isn't that interesting for the testing merging, in my opinion.